modules/component/src/core/utils/emits-synchronously.ts

```diff
@@ -0,0 +1,18 @@
+import { Observable } from 'rxjs';
+
+/**
+ * @description
+ * Checks whether the Observable emits next, error, or complete event synchronously.
+ */
+export function emitsSynchronously(observable$: Observable<unknown>): boolean {
+  let emitsSync = false;
+  observable$
+    .subscribe({
```

Comment:
I think this may lead to unexpected behaviour like side effects that will run twice or may cause performance issue if there is sync cpu heavy computation in stream
IMO value that is received here should be reused as renderedValue

Reply:
Something like this comes to my mind
```typescript
return new Observable<U | undefined | null>(observer => {
        let emitSync = false;
        const subscription = observable$.subscribe({
          next: value => {
            emitSync = true;
            observer.next(value);
          },
          error: err => {
            emitSync = true;
            observer.error(err);
          },
          complete: () => {
            emitSync = true;
            observer.complete();
          },
        });

        // If a new Observable arrives, reset the value to render_creator
        // only if the new Observable doesn't emit the value synchronously
        if (!emitSync) {
          cfg.resetContextObserver.next();
          cfg.render();
        }

        return subscription;
      }).pipe(
        distinctUntilChanged(),
        tap(cfg.updateViewContextObserver),
        tap(() => cfg.render()),
        ...
 ```
 WDYT?

Reply:
Thanks @SebastianPodgajny! The issue with `PushPipe` will be solved with this proposal. However, both solutions won't work properly for `LetDirective`.

Let's try to find a better solution :)